packages/core/src/lib/filter/index.ts

```diff
@@ -253,19 +253,21 @@ class Filter extends BaseProtocol implements IReceiver {
   constructor(libp2p: Libp2p, options?: ProtocolCreateOptions) {
     super(FilterCodecs.SUBSCRIBE, libp2p.components);
 
+    this.pubsubTopics = options?.pubSubTopics || [DefaultPubSubTopic];
+
     libp2p.handle(FilterCodecs.PUSH, this.onRequest.bind(this)).catch((e) => {
       log("Failed to register ", FilterCodecs.PUSH, e);
     });
 
     this.activeSubscriptions = new Map();
-
-    this.options = options ?? {};
   }
 
-  async createSubscription(pubSubTopic?: string): Promise<Subscription> {
-    const _pubSubTopic =
-      pubSubTopic ?? this.options.pubSubTopic ?? DefaultPubSubTopic;
+  async createSubscription(
+    pubSubTopic: string = DefaultPubSubTopic
+  ): Promise<Subscription> {
+    this.ensurePubsubTopicIsValid(pubSubTopic, this.pubsubTopics);
 
+    //TODO: get a relevant peer for the topic/shard
```

Comment:
I guess this will be resolved with `Connection Manager / Discovery` part of the task but maybe mentioning link to the issue will help keep this more clear.